ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -920,3 +923,322 @@ def test_ipa_migrate_stage_mode_with_invalid_cert(self):
         )
         assert result.returncode == 1
         assert ERR_MSG in result.stderr_text
+
+    def test_ipa_migrate_prod_mode_hbac_rule(self, empty_log_file):
+        """
+        This testcase checks that hbac rule is migrated from
+        remote server to local server in prod mode.
+        """
+        hbac_rule_name1 = 'test1'
+        hbac_rule_name2 = 'testuser_sshd'
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name1])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name2])
+        assert hbac_rule_name1 in cmd1.stdout_text
+        assert hbac_rule_name2 in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_sudo_rule(self, empty_log_file):
+        """
+        This testcase checks that sudo cmd and rules are
+        migrated from remote server to local server in prod mode.
+        """
+        sudorule = 'readfiles'
+        sudocmd = '/usr/bin/less'
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "sudorule-find", sudorule])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "sudocmd-find", sudocmd])
+        assert 'Rule name: readfiles\n' in cmd1.stdout_text
+        assert 'Sudo Command: /usr/bin/less\n' in cmd2.stdout_text
+
+    @pytest.fixture
+    def add_remove_user(self):
+        """
+        This fixture adds and remove user required
+        for the test
+        """
+        user_fname = 'produser1'
+        user_lname = 'test'
+        self.master.run_command(
+            [
+                "ipa",
+                "user-add",
+                user_fname,
+                "--first",
+                user_fname,
+                "--last",
+                user_lname,
+            ], raiseonerr=False
+        )
+        yield
+        self.master.run_command(["ipa", "user-del", user_fname])
+
+    def test_ipa_migrate_stage_mode_exist_user_sid(self, add_remove_user):
+        """
+        This testcase checks that in stage-mode uid/gid of the
+        migrated user is not preserved i.e we have different
+        uid/gid for user on remote and local IPA server.
+        """
```

Comment:
I think you need to create users on the replicas[0] before you launch the ipa-migrate command. Otherwise the migration simply picks the next available uid/gid and it will be the same as on the master because you created exactly the same number of entries on both sides.